File: CompositeVisitor/CompositeVisitor_java/Circle.java
public class Circle implements IShape {
  
  private float m_radius;
  
  public Circle(float a_rad) {
    m_radius = a_rad;
  }
  
  public float getArea() {
    return m_radius * m_radius * 3.1415f; 
  }
  
  public void accept(IShapeVisitor a_visitor) {
    a_visitor.visitCircle(this);
  }
}

File: CompositeVisitor/CompositeVisitor_Swift_3.0.1/CompositeVisitor/model/Circle.swift
//
//  Circle.swift
//  CompositeVisitor
//
//  Created by Sunny Buttar on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

class Circle:IShape {
    
    private var m_radius : Float;
    
    init(a_rad : Float) {
        m_radius = a_rad;
    }
    
    func getArea() -> Float {
        return m_radius * m_radius * 3.1415;
    }
    
    func accept(a_visitor: IShapeVisitor) {
        a_visitor.visitCircle(a_circle: self);
    }
}
